api/resource/factory.go

```diff
@@ -124,14 +124,34 @@ func (rf *Factory) SliceFromBytes(in []byte) ([]*Resource, error) {
 	return rf.resourcesFromRNodes(nodes), nil
 }
 
+// DropLocalNodes removes the local nodes by default. Local nodes are detected via the annotation `config.kubernetes.io/local-config: "true"`
+func (rf *Factory) DropLocalNodes(nodes []*yaml.RNode) ([]*Resource, error) {
+	var result []*yaml.RNode
+	for _, node := range nodes {
+		if node.IsNilOrEmpty() {
+			continue
+		}
+		md, err := node.GetValidatedMetadata()
+		if err != nil {
+			return nil, err
+		}
+
+		if rf.IncludeLocalConfigs {
```

Comment:
I noticed this check is duplicated in `KustTarget.IgnoreLocal`. Does this need to be checked twice? Can we just check it in `KustTarget.IgnoreLocal` and not here? 

We can also have it return `[]*yaml.RNode` instead of `[]*Resource`, and we will no longer need the reference to `rf`; this can be a helper function rather than a method on `*Factory`.

Reply:
Right, the IgnoreLocal is also checked in KustTarget.IgnoreLocal.
The reason have IgnoreLocal check here is because
1. DropLocalNodes is a rf function, which is expected to handle the node drop conditions. It will be more error prone to expect its caller function to check IgnoreLocal.
2. DropLocalNodes (previous DropBadNodes) is also used by funcwrappersrc.main, which in the caller function doesn't have the ignoreLocal condition check.

Reply:
I'm not sure if returning []RNode is expected. Actually I used to return []RNode in this function, and realized the kustomize code base is structured in a way that `resource.Factory` handles `RNode`, `kusttarget` is only about resourceaccumulator and ResMap functions but suppose not to know the RNode struct type. `ResMap` (`resmap/factory`) deals with resource but not RNode as well. 
 
Returning []RNode would break this convention and makes the structure harder to maintain.